cms/static/js/views/move_xblock_list.js

```diff
@@ -0,0 +1,181 @@
+        initialize: function(options) {
+            this.visitedAncestors = [];
+            this.template = HtmlUtils.template(MoveXBlockListViewTemplate);
+            this.ancestorInfo = options.ancestorInfo;
+            this.listenTo(Backbone, 'move:backButtonPressed', this.handleBackButtonPress);
+            this.listenTo(Backbone, 'move:breadcrumbButtonPressed', this.handleBreadcrumbButtonPress);
+            this.renderXBlockInfo();
+        },
+
+        render: function() {
+            HtmlUtils.setHtml(
+                this.$el,
+                this.template(
+                    {
+                        xblocks: this.childs_info.childs,
+                        noChildText: this.getNoChildText(),
+                        categoryText: this.getCategoryText(),
+                        XBlocksCategory: this.childs_info.category,
+                        forwardButtonSRText: this.getForwardButtonSRText(),
+                        currentLocationIndex: this.getCurrentLocationIndex()
+                    }
+                )
+            );
+            Backbone.trigger('move:childsInfoRendered', this.breadcrumbInfo());
+            return this;
+        },
+
+        renderChilds: function(event) {
+            this.renderXBlockInfo(
+                'forward',
+                $(event.target).closest('.xblock-item').data('itemIndex')
+            );
+        },
+
+        renderParent: function(newParentIndex) {
+            this.renderXBlockInfo('backward', newParentIndex);
+        },
+
+        handleBackButtonPress: function() {
+            // TODO! improve `this.visitedAncestors.length - 2` mysterious calculation
+            this.renderParent(this.visitedAncestors.length - 2);
+        },
+
+        handleBreadcrumbButtonPress: function(newParentIndex) {
+            this.renderParent(newParentIndex);
+        },
+
+        renderXBlockInfo: function(direction, newParentIndex) {
+            if (direction === undefined) {
+                this.parent_info.parent = this.model;
+            } else if (direction === 'forward') {
+                // clicked child is the new parent
+                this.parent_info.parent = this.childs_info.childs[newParentIndex];
+            } else if (direction === 'backward') {
+                // new parent will be one of visitedAncestors
+                this.parent_info.parent = this.visitedAncestors[newParentIndex];
+                // remove visited ancestors
+                this.visitedAncestors.splice(newParentIndex);
+            }
+
+            this.visitedAncestors.push(this.parent_info.parent);
+
+            if (this.parent_info.parent.get('child_info')) {
+                this.childs_info.childs = this.parent_info.parent.get('child_info').children;
+            } else {
+                this.childs_info.childs = [];
+            }
+
+            this.setDisplayedXBlocksCategories();
+            this.render();
+        },
+
+        setDisplayedXBlocksCategories: function() {
```

Comment:
We are not using `UserFacing` term any where. I will prefer to remain this as it is.